### PR TITLE
Add Sub-jhmdb dataset

### PR DESCRIPTION
**Pull request by jin-s13**

### CODE REVIEW

**Comment by jin-s13** [comment]
Need to upload some model checkpoints, before merging.

File: configs/top_down/cpm/README.md [hunk 1]
```diff
@@ -28,3 +28,26 @@
 | Arch  | Input Size | Mean | Mean@0.1   | ckpt    | log     |
 | :--- | :--------: | :------: | :------: |:------: |:------: |
 | [cpm](/configs/top_down/cpm/mpii/cpm_mpii_368x368.py) | 368x368 | 0.876 | 0.325 | [ckpt](https://download.openmmlab.com/mmpose/top_down/cpm/cpm_mpii_368x368-116e62b8_20200822.pth) | [log](https://download.openmmlab.com/mmpose/top_down/cpm/cpm_mpii_368x368_20200822.log.json) |
+
+
+#### Results on Sub-JHMDB dataset.
+The models are pre-trained on MPII dataset only. NO test-time augmentation (multi-scale /rotation testing) is used.
+
+#####Normalized by Person Size
```
Comment (innerlee) [inline configs/top_down/cpm/README.md:36]: space after #

File: configs/top_down/resnet/README.md [hunk 2]
```diff
@@ -102,3 +102,33 @@ The models are first pre-trained on COCO dataset, and then fine-tuned on PoseTra
 | [pose_resnet_50](/configs/top_down/resnet/posetrack18/res50_posetrack18_256x192.py) | 256x192 | 78.9 | 81.9 | 77.8 | 70.8 | 75.3 | 73.2 | 66.4 | 75.2 | [ckpt](https://download.openmmlab.com/mmpose/top_down/resnet/res50_posetrack18_256x192-a62807c7_20201028.pth) | [log](https://download.openmmlab.com/mmpose/top_down/resnet/res50_posetrack18_256x192_20201028.log.json) |
 
 The models are first pre-trained on COCO dataset, and then fine-tuned on PoseTrack18.
+
+
+#### Results on Sub-JHMDB dataset.
+The models are pre-trained on MPII dataset only. *NO* test-time augmentation (multi-scale /rotation testing) is used.
+
+#####Normalized by Person Size
```
Comment (innerlee) [inline configs/top_down/resnet/README.md:110]: same here

File: README.md [hunk 3]
```diff
@@ -84,6 +84,7 @@ Supported datasets:
 - [x] [AI Challenger](https://github.com/AIChallenger/AI_Challenger_2017)
 - [x] [OCHuman](https://github.com/liruilong940607/OCHumanApi)
 - [x] [CrowdPose](https://github.com/Jeff-sjtu/CrowdPose)
+- [x] [sub-JHMDB](http://jhmdb.is.tue.mpg.de/dataset)
```
Comment (innerlee) [inline README.md:87]: What does `sub-` mean?

File: mmpose/datasets/datasets/top_down/topdown_jhmdb_dataset.py [hunk 4]
```diff
@@ -0,0 +1,347 @@
...
+            [
+                1., 1., 1., 1., 1., 1., 1., 1.2, 1.2, 1.2, 1.2, 1.5, 1.5, 1.5,
+                1.5
+            ],
+            dtype=np.float32).reshape((self.ann_info['num_joints'], 1))
+
+        self.sigmas = np.array([
+            .25, 1.07, .25, .79, .79, 1.07, 1.07, .72, .72, .87, .87, .62, .62,
+            .89, .89
+        ]) / 10.0
```
Comment (innerlee) [inline mmpose/datasets/datasets/top_down/topdown_jhmdb_dataset.py:87]: add note for magic number 10.0

File: mmpose/datasets/datasets/top_down/topdown_coco_dataset.py [hunk 5]
```diff
@@ -204,6 +204,7 @@ def _load_coco_keypoint_annotation_kernel(self, img_id):
                 'image_file': image_file,
                 'center': center,
                 'scale': scale,
+                'bbox': obj['clean_bbox'][:4],
```
Comment (innerlee) [inline mmpose/datasets/datasets/top_down/topdown_coco_dataset.py:207]: Is this a bugfix?

File: mmpose/datasets/datasets/top_down/topdown_jhmdb_dataset.py [hunk 6]
```diff
@@ -0,0 +1,347 @@
...
+                threshold_bbox.append(np.array([bbox_thr, bbox_thr]))
+
+            if 'tPCK' in metrics:
+                torso_thr = np.linalg.norm(item['joints_3d'][4, :2] -
+                                           item['joints_3d'][5, :2])
+                if torso_thr < 1:
+                    torso_thr = np.linalg.norm(
+                        np.array(pred['keypoints'])[4, :2] -
+                        np.array(pred['keypoints'])[5, :2])
+                    print('Torso Size < 1, may be an error....', flush=True)
```
Comment (innerlee) [inline mmpose/datasets/datasets/top_down/topdown_jhmdb_dataset.py:237]: use warning here

**Comment by innerlee** [comment]
Some minor comments

File: README.md (hunk 3 above)
Comment (jin-s13) [reply README.md:87]: Only a subset of JHMDB

File: mmpose/datasets/datasets/top_down/topdown_jhmdb_dataset.py (hunk 4 above)
Comment (jin-s13) [reply mmpose/datasets/datasets/top_down/topdown_jhmdb_dataset.py:87]: Just for better looking..

File: mmpose/datasets/datasets/top_down/topdown_coco_dataset.py (hunk 5 above)
Comment (jin-s13) [reply mmpose/datasets/datasets/top_down/topdown_coco_dataset.py:207]: To make it compatible with JHMDB dataset.

File: README.md (hunk 3 above)
Comment (jin-s13) [reply README.md:87]: JHMDB is initially proposed for action recognition. Some works also use a subset of JHMDB (sub-JHMDB) for pose estimation.

File: mmpose/datasets/datasets/top_down/topdown_jhmdb_dataset.py (hunk 4 above)
Comment (innerlee) [reply mmpose/datasets/datasets/top_down/topdown_jhmdb_dataset.py:88]: haha

Comment (innerlee) [reply mmpose/datasets/datasets/top_down/topdown_jhmdb_dataset.py:88]: not THAT better

File: mmpose/datasets/datasets/top_down/topdown_coco_dataset.py (hunk 5 above)
Comment (innerlee) [reply mmpose/datasets/datasets/top_down/topdown_coco_dataset.py:207]: okay

File: README.md (hunk 3 above)
Comment (innerlee) [reply README.md:87]: Good to know. Put it in the intro of preparation will be helpful

**Review by innerlee — APPROVED** [review]

**Pull request merged**